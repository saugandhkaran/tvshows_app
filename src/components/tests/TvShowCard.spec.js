import { mount } from '@vue/test-utils';
import TvShowCard from '../TVShowCard';

describe('TvShowCard', () => {
  
  it('should match the card component html', () => {
    const wrapper = mount(TvShowCard, {
      propsData: {
        showObject: { image:{ medium :''}}
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('card should render the dummy image image when not in props', () => {
    const wrapper = mount(TvShowCard, {
      propsData: {
        showObject: { image:{ medium :'https://some/url/to/medium/sized/image/jpg'}}
      }
    });
    expect(wrapper.find('[src*="No+image+available"]')).toBeDefined;
    
  });

  it('card should render the image name for accessibility', () => {
    const wrapper = mount(TvShowCard, {
      propsData: {
        showObject: { image:{ medium :'https://some/url/to/medium/sized/image/test.jpg'}, name: 'dummy-alt-image-name'}
      }
    });
    expect(wrapper.find('[alt*="dummy-alt-image-name"]')).toBeDefined;
    
  });

  it('card should render the image when available', () => {
    const wrapper = mount(TvShowCard, {
      propsData: {
        showObject: { image:{ medium :'http://static.tvmaze.com/uploads/images/medium_portrait/215/538786.jpg'}}
      }
    });
    expect(wrapper.find('[src*="medium_portrait/215/538786.jpg"]')).toBeDefined;

  });

  });
