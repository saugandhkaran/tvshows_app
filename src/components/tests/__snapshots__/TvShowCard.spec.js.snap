// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TvShowCard should match the card component html 1`] = `
<article
  class="card"
>
  <div
    class="card-image"
  >
    <img
      src="https://dummyimage.com/210x295/1e1e1e/fff&text=No+image+available"
    />
  </div>
   
  <div
    class="card-heading"
  >
    <p
      class="name"
    >
      
    </p>
     
    <!---->
  </div>
</article>
`;
